C/common/filter_plugin.cpp

```diff
@@ -0,0 +1,242 @@
+void FilterPlugin::shutdown()
+{
+	return this->pluginShutdownPtr(m_instance);
+}
+
+// Ingest readings through filters chain
+void FilterPlugin::ingest(READINGSET* readings)
+{
+        return this->pluginIngestPtr(m_instance, readings);
+}
+
+/**
+ * Load the specified filter plugin
+ *
+ * Static method
+ *
+ * @param    filterName		The filter plugin to load
+ * @return   Plugin handle on success, NULL otherwise 
+ *
+ */
+PLUGIN_HANDLE FilterPlugin::loadFilterPlugin(const string& filterName)
+{
+	if (filterName.empty())
+	{
+		Logger::getLogger()->error("Unable to fetch filter plugin '%s' from configuration.",
+			filterName.c_str());
+		// Failure
+		return NULL;
+	}
+	Logger::getLogger()->info("Loading filter plugin '%s'.", filterName.c_str());
+
+	PluginManager* manager = PluginManager::getInstance();
+	PLUGIN_HANDLE handle;
+	if ((handle = manager->loadPlugin(filterName, PLUGIN_TYPE_FILTER)) != NULL)
+	{
+		// Suceess
+		Logger::getLogger()->info("Loaded filter plugin '%s'.", filterName.c_str());
+	}
+	return handle;
+}
+
+/**
+ * Cleanup all the load filters setup
+ *
+ * Call "plugin_shutdown" method and free the FilterPlugin object
+ *
+ * Static method
+ *
+ * @param loadedFilters		The vector of loaded filters
+ *
+ */
+void FilterPlugin::cleanupFilters(std::vector<FilterPlugin *>& loadedFilters)
+{
+	// Cleanup filters
+	for (auto it = loadedFilters.begin(); it != loadedFilters.end(); ++it)
+	{
+		// Call filter plugin shutdown
+		(*it)->shutdown();
+		// Free filter
+		delete *it;
+	}
+}
+
+/**
+ * Load all filter plugins found in the configuration category
+ *
+ * Static method
+ *
+ * @param categoryName	Configuration category
+ * @param filters	Vector of FilterPlugin to be filled
+ * @param manager	The management client
+ * @return		True if filters are loaded (or no filters at all)
+ *			False otherwise
+ */
+bool FilterPlugin::loadFilters(const string& categoryName,
+			       std::vector<FilterPlugin *>& filters,
+			       ManagementClient* manager)
+{
+        try
+        {
+        	// Get the category with values and defaults
+        	ConfigCategory config = manager->getCategory(categoryName);
+                string filter = config.getValue("filter");
+                if (!filter.empty())
+                {
+                       map<string, PLUGIN_HANDLE> filterInfo = {};
+
+			// Remove \" and leading/trailing "
+			// TODO: improve/change this
+			filter.erase(remove(filter.begin(), filter.end(), '\\' ), filter.end());
+			size_t i;
+			while (! (i = filter.find('"')) || (i = filter.rfind('"')) == static_cast<unsigned char>(filter.size() - 1))
+			{
+				filter.erase(i, 1);
+			}
+
+			//Parse JSON object for filters
+			Document theFilters;
+			theFilters.Parse(filter.c_str());
+			// The "pipeline" property must be an array
+			if (theFilters.HasParseError() ||
+				!theFilters.HasMember("pipeline") ||
+				!theFilters["pipeline"].IsArray())
+			{
+				throw runtime_error("Cannot parse JSON 'filter' property");
```

Comment:
Could you log something here as well, the config for this could get complex so having as much feedback as we can would be good.

Reply:
yes, missing log message added